Make the generate_homepage script more generic.

whereistanya opened the pull request:
Generate a page from a yaml config.

whereistanya committed:
Make generate_homepage_from_content generic.

Move from csv to yaml, and move text elements out of the html file. Also add a
README.

whereistanya committed:
Add config for the earlier index.html. Print unicode.

Add in the tweets listed in this version of the index.html:

https://github.com/DevProgress/not-my-locker-room/blob/4e33262ccc8031b3703b1f9d33eaa980aa7e6a75/index.html

The HTML generated by the script differs from that file, but it looks (to me)
the same when rendered.

whereistanya committed:
Allow overriding image. Add 'endorsements' example.

whereistanya committed:
Explicitly add image to the locker-room content.

maiamcc commented on generate_homepage_from_content.py:
```diff
@@ -147,11 +150,15 @@ def html_element_from_content(content_dict):
+      try:
+        contents = yaml.load(stream)
+      except yaml.YAMLError as ex:
+        print ex
```
I don't see much point to catching this error if we don't add any additional info to it OR break the func. I would think we would WANT to `raise` the exception rather just printing the exception and potentially returning nonsense--which will either lead to more, weirder errors down the line, or us generating a blank content page and possibly overwriting a file.

I'd recommend just not catching the error--if something goes wrong, it'll raise an exception and crash the program--or if you are gonna catch it, just do so to add additional info (human readable string like "Something went wrong trying to read content from the yaml file!") and `raise` the exception anyway.

whereistanya replied:
Good point. Removed the try/except entirely.

maiamcc commented on generate_homepage_from_content.py:
```diff
@@ -27,7 +29,8 @@
 WEBSITE_CONTENT_TEMPLATE = ('<span class="website-content">%s</span>\n'
-                            '<span class="website-url">%s</span>')
+                            '<span class="website-url">'
+                            '<a href="%s">%s</a></span><br><br>')
```
This `span` element may have been changed to `div` in @jillh510's latest diff, so maybe a moot point...? But regardless, I'd say, don't use `<br>` and accomplish the spacing with css styling instead.

whereistanya replied:
I'll take out the line breaks and wait for Jill's change. And actually, I see she linkifies the url as well, so I'll remove that from mine to make a cleaner merge.

maiamcc reviewed (approved):
Awesome stuff! Left a few small comments but when you fix those, you're good to merge as far as I'm concerned! (Unless @jillh510 also wants to take a look?)

whereistanya committed:
Don't catch exception if the yaml load fails; just die.

whereistanya committed:
Remove linkification of webpages for now.

Jill's already got it in
https://github.com/DevProgress/not-my-locker-room/pull/8
and I think removing it here should give us a conflict free merge.
